daml-assistant/daml-project-config/DAML/Project/Consts.hs

```diff
@@ -132,18 +138,26 @@ getSdkVersion = getEnv sdkVersionEnvVar
 getDamlAssistant :: IO FilePath
 getDamlAssistant = getEnv damlAssistantEnvVar
 
+-- | Whether we should check if a command is invoked inside of a project.
+-- The string is the command name used in error messages
+data ProjectCheck = ProjectCheck String Bool
```

Comment:
I'm not a fan of bringing this into daml-project-config, since this is very much tied with damlc command-line option parsing.

Reply:
Is it? I would argue that having an argument that enforces that `withProjectRoot` is run in a project and errors out if that is not the case is quite reasonable even if you completely ignore the CLI option parsing. The `String` argument might be a bit tied to details in damlc but it seems sufficiently general that I believe that this API would be useful for any tool invoked by the assistant so `daml-project-config` is not a bad place for this.

Reply:
Ok, that makes sense. :+1: